lamindb/models/artifact.py

```diff
@@ -1411,9 +1409,17 @@ def __init__(
             return None
         # now proceed with the user-facing constructor
         if len(args) > 1:
-            raise ValueError("Only one non-keyword arg allowed: data")
+            raise ValueError("Only one non-keyword arg allowed: path")
 
-        data: str | Path = kwargs.pop("data") if len(args) == 0 else args[0]
+        if "data" in kwargs:
+            warnings.warn(
+                "`data` argument will be renamed to `path` in a future release.",
```

Comment:
But it's already been renamed here, right? It will be removed in a future release.

```suggestion
                "`data` argument was renamed to `path` and will be removed in a future release.",
```

so something like this?

Reply:
Yes, thanks for spotting! This was tab complete from AI ☺️